library/spdm_crypt_lib/libspdm_crypt_asym.c

```diff
@@ -526,6 +526,144 @@ static bool libspdm_asym_func_need_hash(uint32_t base_asym_algo)
     return false;
 }
 
+
+static void libspdm_copy_signature_swap_endian_rsa(
+    uint8_t* dst,
+    size_t dst_size,
+    const uint8_t* src,
+    size_t src_size)
+{
+    /* RSA signature is a single buffer to be swapped */
+    size_t i;
+
+    if (src == dst) {
+        LIBSPDM_ASSERT(dst_size == src_size);
+
+        /* src and dst are same buffer. Swap in place. */
+
+        uint8_t byte;
+        for (i = 0; i < dst_size / 2; i++) {
+            byte = dst[i];
+            dst[i] = dst[dst_size - i - 1];
+            dst[dst_size - i - 1] = byte;
+        }
+    }
+    else {
+        /* src and dst are different non-overlapping buffers */
+        LIBSPDM_ASSERT(dst_size >= src_size);
+        LIBSPDM_ASSERT((src < dst && src + src_size <= dst) ||
+                       (dst < src && dst + dst_size <= src));
+
+        for (i = 0; i < src_size; i++) {
+            dst[i] = src[src_size - i - 1];
+        }
+    }
+}
+
+static void libspdm_copy_signature_swap_endian_ecdsa(
+    uint8_t* dst,
+    size_t dst_size,
+    const uint8_t* src,
+    size_t src_size)
+{
+    /* ECDSA signature is actually 2 buffers (x & y)
+     * and each must be swapped individually */
+    size_t i;
+
+    if (src == dst) {
+        LIBSPDM_ASSERT(dst_size == src_size);
+
+        /* src and dst are same buffer. Swap ecdsa 2 internal buffers in place. */
+
+        size_t x_size;
+        size_t y_size;
+        uint8_t* x;
+        uint8_t* y;
+        uint8_t byte;
+
+        x_size = dst_size / 2;
+        y_size = x_size;
+
+        x = dst;
+        y = x + x_size;
+
+        for (i = 0; i < x_size / 2; i++) {
+            byte = x[i];
+            x[i] = x[x_size - i - 1];
+            x[x_size - i - 1] = byte;
+        }
+
+        for (i = 0; i < y_size / 2; i++) {
+            byte = y[i];
+            y[i] = y[y_size - i - 1];
+            y[y_size - i - 1] = byte;
+        }
+    }
+    else {
+        /* src and dst are different non-overlapping buffers */
```

Comment:
`!(src == dst)` doesn't mean that they are non-overlapping.

Reply:
can we just swap in place?

Reply:
See same comment earlier.